wom/src/main/scala/wom/types/WomFileType.scala

```diff
@@ -1,56 +1,69 @@
 package wom.types
 
 import spray.json.JsString
-import wom.values.{WomFile, WomGlobFile, WomSingleFile, WomString}
+import wom.values.{WomGlobFile, WomSingleDirectory, WomSingleFile, WomString}
 
 import scala.util.{Success, Try}
 
-sealed abstract class WomFileType extends WomPrimitiveType
+sealed trait WomFileType extends WomPrimitiveType {
+  // TODO: WOM: WOMFILE: When WDL supports directories this check may need more work (refactoring to subclasses?)
+  override def equals(rhs: WomType): Try[WomType] = rhs match {
+    case _: WomFileType => Success(WomBooleanType)
+    case WomStringType => Success(WomBooleanType)
+    case WomOptionalType(memberType) => equals(memberType)
```

Comment:
Wouldn't this say that `WomFileType equals WomOptionalType(WomFileType)` ? Unless that's the intended behaviour ?

Reply:
This lets you say:
```wdl
Int? input
if (input == 5) { ... }
```